realm/realm-annotations-processor/src/main/java/io/realm/processor/ClassMetaData.java

```diff
@@ -329,12 +329,15 @@ public String getPrimaryKeyGetter() {
         return getGetter(primaryKey.getSimpleName().toString());
     }
 
+    /**
+     * Checks if a VariableElement is nullable. In case of PrimaryKey field, such field can be
+     * nullable only if its type could be found in {@link Constants#JAVA_TO_NULLABLE_PRIMARYKEY_TYPES}.
+     *
+     * @return {@code true} if a VariableElement is nullable type, {@code false} otherwise.
+     */
     public boolean isNullable(VariableElement variableElement) {
-        // primary keys cannot be nullable
-        if (hasPrimaryKey()) {
-            if (variableElement.equals(getPrimaryKey())) {
-                return false;
-            }
+        if (hasPrimaryKey() && variableElement.equals(getPrimaryKey())) {
```

Comment:
`ClassMetaData.isNullable()` is used many times in `RealmProxyClassGenerator` to check if only  _said_ types are allowed to accept `null` as @PrimaryKey value. 

The two previous conditions are combined into one-liner and then checks if a `VariableElement` could be nullable with the help of `Constants#JAVA_TO_NULLABLE_PRIMARYKEY_TYPES`.

Somewhere, somehow, we need to check `VariableElement` is allowed to be nullable and I thought this is a neat place to put the checker logic in with minimal change to the codebase.